WcaOnRails/app/views/static_pages/about.html.erb

```diff
@@ -1,101 +1,7 @@
-      <%= t('about.donation_banner.donate_paypal') %>
-    </a>
-    <a class="btn btn-primary btn-lg" href="#contribute" role="button"><%= t('about.donation_banner.donate_other') %></a>
-  </div>
-
-  <blockquote class="blockquote">
-    <p class="mb-0"><%= t('about.mission_statement') %></p>
-    <footer class="blockquote-footer"><%= t('about.mission_footer') %></footer>
-  </blockquote>
-
-  <h2><%= t('about.who_we_are_title') %></h2>
-  <div class="row">
-    <div class="col-md-8">
-    <%= t('about.who_we_are_content_html') %>
-    </div>
-    <div class="col-md-4">
-      <img class="about-page-image" src=<%= asset_url("about/bulgarian-open.jpg") %> />
-    </div>
-  </div>
-
-  <blockquote class="blockquote">
-    <p class="mb-0"><%= t('about.spirit') %></p>
-    <footer class="blockquote-footer"><%= t('about.spirit_footer') %></footer>
-  </blockquote>
-
-
-  <h2><%= t('about.our_goals_title') %></h2>
-
-  <%= t('about.our_goals_content_html') %>
-
-  <h2><%= t('about.whats_next_title') %></h2>
-  <%= t('about.whats_next_content_html') %>
-
-  <h2><%= t('about.world_championship_title') %></h2>
-  <div class="row">
-    <div class="col-md-8">
-    <%= t('about.world_championship_content_html') %>
-      </div>
-    <div class="col-md-4">
-      <img class="about-page-image" src=<%= asset_url("about/worlds-2015.jpg") %> />
-    </div>
-  </div>
-
-  <h2><%= t('about.structure.title') %></h2>
-
-  <p><%= t('about.structure.board.description') %></p>
-
-  <ul>
-    <% Team.board.current_members.includes(:user).map(&:user).sort_by(&:name).each do |board_member| %>
-      <li><%= board_member.name %> - <%= board_member.country.name %></li>
-    <% end %>
-  </ul>
-
-  <p>
-    <%= t 'about.structure.operations_html', committees_and_teams: link_to(t('about.structure.teams_committees'), teams_committees_path) %>
-  </p>
-
-  <p>
-    <%= t 'about.structure.delegates_html', see_link: t('about.structure.see_link_html', link: link_to(t('delegates_page.title'), delegates_path)) %>
-  </p>
-
-  <p>
-    <%= t('about.structure.members_html') %>
-  </p>
-
-  <h2 id="contribute"><%= t('about.contributing_title') %></h2>
-  <p>
-  <%= t('about.contributing_content_html',
-        here_credit: link_to(t('common.here'), StaticPagesController::DONATE_GUIDESTAR_URL),
-        here_paypal: link_to(t('common.here'), StaticPagesController::DONATE_PAYPAL_URL)) %>
-    <em>
-      <br />
-      <br />
-      World Cube Association
-      <br />
-      5042 Wilshire Blvd #43206
-      <br />
-      Los Angeles, CA 90036
-      <br />
-      United States of America
-      <br /><br />
-    </em>
-    <%= t('about.tax_deductible_note') %>
-  </p>
-
-  <h2><%= t('about.other_help_title') %></h2>
-  <%= t('about.other_help_content_html') %>
-
-</div>
+<%= react_component('StaticPages/About', {
+  currentBoardMembers: Team.board.current_members.includes(:user).map(&:user).sort_by(&:name)
```

Comment:
More of a general question, rather than for this specific PR, but I'm a little unclear for React components on when we should pass in user/competition/incident/etc. data as params, and when we should fetch it from an API endpoint.

Reply:
Yes I was wondering exactly the same thing actually!

I only made it take the data as a prop as that option is much easier to implement, but I do think that this case would be a perfect opportunity to create a new API endpoint.
I'd be happy to make the endpoint though if others agree?

Reply:
I've added my two cents at https://docs.google.com/document/d/1X7veopxmMFKJtXT7WFpcn0w6MqW68fBybSlvBf25bYY/edit#heading=h.w4hztc87n5so

Reply:
In this specific case, I don't see a real value to creating an endpoint because the Board members never change once the page has been loaded.

Reply:
(I can't access the doc, sorry if my opinion is already reflected there)
IMO basically components should be written as if they were getting their data from "somewhere" not caring about where that somewhere is (except for the component actually doing the data fetch obviously :)); it will make it easier to re-use them if/when the data provider change.

As for static pages, I think providing the data directly in the react component render helper is more helpful: if I'm not mistaken it would also let us enable server side rendering, which is always something good to consider to help search engines.

Reply:
The only thing I'd like to point out is that this component is only used once, and the same data is being passed to the component every time. I can't see a case where you would pass any other data to the component, which means that surely the data should be entirely handled within the component? I feel like in cases like this where, wherever you use it, the same data is required, the component should be responsible for getting that data, rather than the parent component (or Rails) passing the same data down every time.

Having said that, using an API endpoint is harder to implement, and I am completely happy to leave it as is.

Reply:
I don't see an advantage in having the page fetch its own data if that data never changes after the initial fetch.
Do you have a concrete reason for your feeling that the component should be responsible for getting that data?

Reply:
> The only thing I'd like to point out is that this component is only used once, and the same data is being passed to the component every time. I can't see a case where you would pass any other data to the component, which means that surely the data should be entirely handled within the component?

For the record I was commenting on the general approach for creating a component, in this specific case it's clear that with data changing once every year or so it's not quite efficient to make an API call for it, and that passing them through the rails rendering helper should be the way to go.

Reply:
I don't doubt that it would be much more performant to pass this very small amount of data as a prop, however I don't feel that the UX would be impacted at all regardless of what approach is used.

I was more thinking purely from a design choice perspective, and what parts of the page should be responsible for what data. It just feels neater if the component can be placed into any page without that page having to worry about giving data to the component, data which is always the same regardless of where the component is (that is as opposed to a button, say, which could be rendered with any text in it).